src/main/java/net/javadiscord/javabot/data/h2db/message_cache/dao/MessageCacheRepository.java

```diff
@@ -0,0 +1,164 @@
+package net.javadiscord.javabot.data.h2db.message_cache.dao;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.javadiscord.javabot.data.h2db.message_cache.model.CachedMessage;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Dao class that represents the QOTW_POINTS SQL Table.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class MessageCacheRepository {
+	private final Connection con;
+
+	/**
+	 * Inserts a new {@link CachedMessage} object.
+	 *
+	 * @param message The Message to insert.
+	 * @return Whether there were rows affected by this process.
+	 * @throws SQLException If an error occurs.
+	 */
+	public boolean insert(CachedMessage message) throws SQLException {
+		PreparedStatement stmt = con.prepareStatement("INSERT INTO message_cache (message_id, author_id, message_content) VALUES (?, ?, ?)",
+				Statement.RETURN_GENERATED_KEYS
+		);
+		stmt.setLong(1, message.getMessageId());
+		stmt.setLong(2, message.getAuthorId());
+		stmt.setString(3, message.getMessageContent());
+		int rows = stmt.executeUpdate();
+		stmt.close();
+		return rows > 0;
+	}
+
+	/**
+	 * Inserts a {@link List} of {@link CachedMessage} objects..
+	 *
+	 * @param messages The List to insert.
+	 * @return Whether there were rows affected by this process.
+	 * @throws SQLException If an error occurs.
+	 */
+	public boolean insertList(List<CachedMessage> messages) throws SQLException {
+		StringBuilder statementString = new StringBuilder("INSERT INTO message_cache (message_id, author_id, message_content) VALUES");
+		for (CachedMessage msg:messages) {
+			statementString.append(String.format(" (%s, %s, '%s'),", msg.getMessageId(), msg.getAuthorId(), msg.getMessageContent()));
+		}
+		statementString.deleteCharAt(statementString.toString().length() - 1).append(";");
+		PreparedStatement stmt = con.prepareStatement(statementString.toString(),
+				Statement.RETURN_GENERATED_KEYS
+		);
+		int rows = stmt.executeUpdate();
+		stmt.close();
+		return rows > 0;
+	}
```

Comment:
Does this even work? Apending the values seems weird, you could just loop over all CachedMessage objects and call the single insert method instead.
But if this work, ig i'm fine with that

Reply:
I'm appending them because there's no way to insert multiple values of varying amount in a single update with prepared statements

Why would I loop through them and insert them one by one, that just uses up unnecessary resources and time

Reply:
aight, if that works

Reply:
Nah, I just put that there without testing 😉

`INSERT INTO message_cache (message_id, author_id, message_content) VALUES (xxx, xxx, xxx), (xxx, xxx, xxx), (xxx,xxx,xxx);`
Is just how you insert multiple rows with one statement 👍

Reply:
Oh yeah bruh, thought you overwrite the values each time, nvm then lol